samcli/local/services/base_local_service.py

```diff
@@ -118,4 +121,19 @@ def get_lambda_output(stdout_stream):
             # Last line is Lambda response. Make sure to strip() so we get rid of extra whitespaces & newlines around
             lambda_response = stdout_data[last_line_position:].strip()
 
-        return lambda_response, lambda_logs
+        is_lambda_user_error_response = False
+
+        try:
+            lambda_response_dict = json.loads(lambda_response)
```

Comment:
Also, you need to decode the lambda_response before json.loads it. Might be worth moving the decode to a upstream call somewhere. See this comment: https://github.com/awslabs/aws-sam-cli/issues/531#issuecomment-402998251